pages/views/auth.py

```diff
@@ -350,4 +351,35 @@ def passwordResetEmail(request, id, feedback):
     passwordResetFeedback.completed = True
     passwordResetFeedback.save()
 
-    return HttpResponse(str(msgBody))
\ No newline at end of file
+    return HttpResponse(str(msgBody))
+
+def usernameLookup(request):
+    resources = ResourceLink.objects.all()
+
+    if request.method == "POST":
+        form = UsernameLookupForm(request.POST)
+
+        if form.is_valid():
+            formEmail = form.cleaned_data['email']
+            userMatchingEmail = authUser.objects.filter(email=formEmail)
+
+            if userMatchingEmail.exists:
+                for user in userMatchingEmail:
+                    if check_password(form.cleaned_data['password'], user.password):
+                        cftsUser = User.objects.get(auth_user=user)
+                        usernameLookupFeedback = Feedback(title="Username Lookup: " + str(user.last_name) + ", " + str(user.first_name) + "(" + str(user.email) + ")", user=cftsUser, category="Username Lookup")
```

Comment:
- No, bad!
- We don't need to generate feedback or send emails for this
- If we can filter down a user based on email and the the provided password matches just show them the username on page

Reply:
Changed - whiner.